flax/linen/module.py

```diff
@@ -1157,11 +1158,11 @@ def _module_checks(self):
       raise errors.IncorrectPostInitOverrideError()
 
   @traceback_util.api_boundary
-  def bind(self,
+  def bind(self: M,
```

Comment:
The description "Creates an interactive Module instance by binding variables and RNGs." is a bit odd though, maybe we should just call it a "stateful Module instance", and say it can be used interactivelly?

Reply:
@marcvanzee I think we should do this in another PR. Want me to create an separate issue for this?

Reply:
👍